packages/api/src/pins.js

```diff
@@ -106,7 +103,7 @@ export async function pinPost (request, env, ctx) {
 
   // Validate cid
   try {
-    normalizeCid(cid)
+    pinData.normalizedCid = normalizeCid(cid)
```

Comment:
Personally I'd not augment this user provided data with anything but instead pull out the bits you want and combine them any derived/additional data in a different object.

Reply:
You caught me, I was being lazy 😛. On it.